torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3955,6 +3955,48 @@ def sample_inputs_matmul(op_info, device, dtype, requires_grad):
     return tuple(sample_inputs)
 
 
+def sample_inputs_meshgrid(
+        calling_convention: str) -> Callable[[OpInfo, torch.device, torch.dtype, bool], List[SampleInput]]:
+    if calling_convention == 'variadic':
+        def make_inputs(
+                tensors: List[torch.Tensor]) -> Tuple[Union[torch.Tensor,
+                                                            List[torch.Tensor]],
+                                                      Tuple[torch.Tensor, ...]]:
+            return tensors[0], tuple(tensors[1:])
+    elif calling_convention == 'list':
+        def make_inputs(
+                tensors: List[torch.Tensor]) -> Tuple[Union[torch.Tensor,
+                                                            List[torch.Tensor]],
+                                                      Tuple[torch.Tensor, ...]]:
+            return tensors, ()
+    else:
+        raise ValueError(
+            'Unsupported calling convention, must be one of {"variadic", "list"}. '
+            f'Got "{calling_convention}".')
+
+    def sample_inputs(op_info: OpInfo, device: torch.device, dtype: torch.dtype,
+                      requires_grad: bool) -> List[SampleInput]:
+        Z = ()  # note: torch.rand(()) yields a random 0D tensor
```

Comment:
`Z` seems like a fine addition to the `S` `M` and `L` defined at the global scope

Reply:
done. i have some concern that `Z` is not necessarily interchangeable with the other sizes, since it represents a full shape, whereas sometimes `S` et al. are used as the length of a single dimension.

Reply:
Fair point, but from that perspective I'd suggest simply not defining it and always writing `()`, then, because that concern is also a local concern

Reply:
True, it is a local concern. `S` here is also a local, not the value from above. Do you think that having two local constants `S` and `Z` here would be inappropriate?

```
S = torch.Size([3])
Z = torch.Size([])
```

Reply:
I'd prefer not shadowing the global for readability, especially not shadowing it with a very similar value

Reply:
changed to locals named SCALAR and VECTOR